### PR TITLE
Add a verifier for bgv.rotate and corresponding test (#287)

### PR DESCRIPTION
**Pull request by Maokami**
- add verifier for bgv.rotate
- add simple expected-error-test for bgv.rotate

### CODE REVIEW

File: tests/bgv/verifier.mlir [hunk 1]
```diff
@@ -0,0 +1,13 @@
+// RUN: heir-opt --verify-diagnostics --split-input-file %s
```
Comment (j2kun) [inline tests/bgv/verifier.mlir:1]: Nit: `--split-input-file` is only necessary if you have multiple tests in the same file that must be parsed as if they were separate files (it's a weird convention), and the separator to mark when to split the file is `// -----`

File: lib/Dialect/BGV/IR/BGVDialect.cpp [hunk 2]
```diff
@@ -53,6 +53,17 @@ LogicalResult MulOp::verify() {
   }
   return success();
 }
+LogicalResult Rotate::verify() {
+  auto x = this->getX().getType();
```
Comment (j2kun) [inline lib/Dialect/BGV/IR/BGVDialect.cpp:57]: nit: I know the rest of this file uses `this->`, but it should not be necessary (here or in the rest of the file). I think you can just delete `this->` everywhere and it should run the same.

Context: Generated Op classes have this strange property that they are treated both as their own class (with named arguments to the fields created via tablegen) and simultaneously as a pointer to `Operation`, which has methods that generically apply to all ops, and can do things like mutate the operation in place.

**Review by j2kun — APPROVED** [review]
Looks great, thanks for the contribution! I added two comments for optional improvements. If you don't have time for those I can go in and fix them after it merges.

File: tests/bgv/verifier.mlir (hunk 1 above)
Comment (Maokami) [reply tests/bgv/verifier.mlir]: Applied!

File: lib/Dialect/BGV/IR/BGVDialect.cpp (hunk 2 above)
Comment (Maokami) [reply lib/Dialect/BGV/IR/BGVDialect.cpp]: Applied!

**Review by j2kun — APPROVED** [review]

**Pull request merged**